app/controllers/cloud_volume_snapshot_controller.rb

```diff
@@ -36,21 +36,12 @@ def specific_buttons(pressed)
   def delete_cloud_volume_snapshots
     assert_privileges("cloud_volume_snapshot_delete")
 
-    snapshots = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "cloud_volume_snapshot")
-                  find_checked_ids_with_rbac(CloudVolumeSnapshot)
-                else
-                  [find_id_with_rbac(CloudVolumeSnapshot, params[:id])]
-                end
-
-    if snapshots.empty?
-      add_flash(_("No Cloud Volume Snapshots were selected for deletion."), :error)
-    end
+    snapshots = find_records_with_rbac(CloudVolumeSnapshot, checked_or_params)
 
     snapshots_to_delete = []
-    snapshots.each do |snapshot_id|
-      snapshot = CloudVolumeSnapshot.find_by_id(snapshot_id)
-      if snapshot.nil?
-        add_flash(_("Cloud Volume Snapshot no longer exists."), :error)
+    snapshots.each do |snapshot|
+      if !snapshot.supports?(:delete)
```

Comment:
@JakubKubista this is still incorrect. There was no test for the records support before. You're changing the behaviour of the method.

The whole block operating with `snapshots_to_delete` can be removed, as well as the final test `unless snapshots_to_delete.empty?`

The only thing tested in the block is a presence of a records in `snapshots` variable, and we already know, there always has to be something in the result of `find_records_with_rbac` method